### PR TITLE
REFACTOR-#2059: Consolidate environment variables

### LINKED ISSUE #2059: Group all controlling environment variables in single file
**Is your feature request related to a problem? Please describe.**
Currently it's hard both for Modin developer and for end user to see which variables exist and what they do.
I propose to move all variables to a single unified file and replace their usage with getter functions which will have docstrings explaining what variables do.

### PR DESCRIPTION
**Pull request by vnlitvinov**
## What do these changes do?
Consolidate all environment variables used for Modin configuration in single file.
Also introduce the API to manage configuration, could be generalized later for non-envinronment-based options.

Also provide short help on existing options by running `python -m modin.config`.
<details>
<summary>Current help</summary>

```
MODIN_BACKEND: Engine running on a single node of distribution
        Provide a string (valid examples are: Pandas, OmniSci, Pyarrow)
        Current value: Pandas
MODIN_CPUS: How may CPU cores to utilize across the whole distribution
        Provide an integer value
        Current value: 8
MODIN_LOG_RPYC: Whether to gather RPyC logs (applicable for remote context)
        Provide a boolean flag (any of 'true', 'yes' or '1' in case insensitive manner is considered positive)
        Current value: None
MODIN_TRACE_RPYC: Whether to trace RPyC calls (applicable for remote context)
        Provide a boolean flag (any of 'true', 'yes' or '1' in case insensitive manner is considered positive)
        Current value: None
MODIN_USE_CALCITE: Whether to use Calcite for OmniSci queries execution
        Provide a boolean flag (any of 'true', 'yes' or '1' in case insensitive manner is considered positive)
        Current value: None
MODIN_ENGINE: Distribution engine to run queries by
        Provide a string (valid examples are: Ray, Dask, Python)
        Current value: Ray
MODIN_DEBUG: Forces Modin engine to be "Python" unless specified by $MODIN_ENGINE
        Provide a boolean flag (any of 'true', 'yes' or '1' in case insensitive manner is considered positive)
        Current value: None
MODIN_EXPERIMENTAL: Turns on experimental features
        Provide a boolean flag (any of 'true', 'yes' or '1' in case insensitive manner is considered positive)
        Current value: None
MODIN_OUT_OF_CORE: Unknown
        Provide a boolean flag (any of 'true', 'yes' or '1' in case insensitive manner is considered positive)
        Current value: None
MODIN_RAY_CLUSTER: True if Modin is running on pre-initialized Ray cluster
        Provide a boolean flag (any of 'true', 'yes' or '1' in case insensitive manner is considered positive)
        Current value: None
MODIN_MEMORY: How much memory give to each Ray worker (in bytes)
        Provide an integer value
        Current value: None
MODIN_OMNISCI_FRAGMENT_SIZE: How big a fragment in OmniSci should be when creating a table (in rows)
        Provide an integer value
        Current value: None
MODIN_ON_RAY_PLASMA_DIR: Path to Plasma storage for Ray
        Provide a string
        Current value: None
MODIN_REDIS_ADDRESS: What Redis address to connect to when running in Ray cluster
        Provide a string
        Current value: None
MODIN_SOCKS_PROXY: SOCKS proxy address if it is needed for SSH to work
        Provide a string
        Current value: None
MODIN_TEST_DATASET_SIZE: Dataset size for running some tests
        Provide a string (valid examples are: small, normal, big)
        Current value: None
```

</details>

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2059 
- [x] tests added and passing

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -76,6 +76,29 @@ jobs:
...
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-python-3.7-pip-${{ github.run_id }}-${{ hashFiles('environment.yml') }}
+      - uses: goanpeca/setup-miniconda@v1.6.0
+        with:
+          activate-environment: modin
+          environment-file: environment.yml
+          python-version: 3.7
+          channel-priority: strict
+          use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly!
```
Comment (anmyachev) [inline .github/workflows/ci.yml:98]: Is it mandatory for the `pylint` to create a complete dev environment?

File: modin/config/envvars.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+class RayPlasmaDir(EnvironmentVariable, type=str):
+    """
+    Path to Plasma storage for Ray
+    """
+
+    varname = "MODIN_ON_RAY_PLASMA_DIR"
+
+
+class IsOutOfCore(EnvironmentVariable, type=bool):
```
Comment (anmyachev) [inline modin/config/envvars.py:153]: Missing class-level docstring

File: modin/config/envvars.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+class TestDatasetSize(EnvironmentVariable, type=str):
+    """
+    Dataset size for running some tests
+    """
+
+    varname = "MODIN_TEST_DATASET_SIZE"
+    choices = ("small", "normal", "big")
+
+
+def _check_vars():
```
Comment (anmyachev) [inline modin/config/envvars.py:206]: add docstring please

File: modin/config/envvars.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+class Engine(EnvironmentVariable, type=str):
+    """
+    Distribution engine to run queries by
+    """
+
+    varname = "MODIN_ENGINE"
+    choices = ("Ray", "Dask", "Python")
+
+    @classmethod
+    def _get_default(cls):
```
Comment (anmyachev) [inline modin/config/envvars.py:58]: add docstring please

File: modin/config/envvars.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    """
+
+    varname: str = None
+
+    @classmethod
+    def _get_raw_from_config(cls) -> str:
+        return os.environ[cls.varname]
+
+    @classmethod
+    def get_help(cls) -> str:
```
Comment (anmyachev) [inline modin/config/envvars.py:34]: Maybe add some short example of help string?

**Review by anmyachev — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (vnlitvinov) [reply .github/workflows/ci.yml]: I'm removing `pylint` section for now, it should be a separate PR.

File: modin/config/envvars.py (hunk 2 above)
Comment (vnlitvinov) [reply modin/config/envvars.py:153]: I know, but I have no clue what this variable does :) @devin-petersohn could you pls help?

Comment (anmyachev) [reply modin/config/envvars.py:153]: Maybe this can be useful https://modin.readthedocs.io/en/latest/out_of_core.html

Comment (vnlitvinov) [reply modin/config/envvars.py:153]: nice find, thx, docstring added

File: modin/config/envvars.py (hunk 4 above)
Comment (vnlitvinov) [reply modin/config/envvars.py:58]: added

File: modin/config/envvars.py (hunk 5 above)
Comment (vnlitvinov) [reply modin/config/envvars.py:34]: I don't see a need for it. The output is provided in PR description if you expand the spoiler, no need IMHO to insert it into the source.

File: modin/config/envvars.py (hunk 3 above)
Comment (vnlitvinov) [reply modin/config/envvars.py:210]: added

**Comment by anmyachev** [comment]
LGTM, let's wait @devin-petersohn.

**Review by anmyachev — DISMISSED** [review]

**Comment by vnlitvinov** [comment]
Anything needed from me to get this merged?

**Comment by anmyachev** [comment]
> Anything needed from me to get this merged?

I think nothing is needed.

**Comment by anmyachev** [comment]
@vnlitvinov please restart CI. Tests for Omnisci backend should be working now.

**Comment by anmyachev** [comment]
After this I will do merge.

**Comment by vnlitvinov** [comment]
I've rebased off `master` to make sure I got all recent CI fixes

**Comment by vnlitvinov** [comment]
@anmyachev @devin-petersohn I believe the failure in running OmniSci tests isn't related to this PR (the tests actually pass, all the things crashed and burnt during the exit of the interpreter according to the logs).

I think we should merge this PR.

**Review by devin-petersohn — APPROVED** [review]
LGTM, it's quite clean.

Can we create an issue to add this list to the documentation?

**Pull request merged**